ncm-pcs/src/main/pan/components/pcs/schema.pan

```diff
@@ -0,0 +1,163 @@
+${componentschema}
+
+include 'quattor/types/component';
+include 'pan/types';
+
+type pcs_score = string with match(SELF, '^(-?\d+|-?INFINITY)$');
+
+type pcs_cluster = {
+    @{name of the cluster}
+    'name' : string
+    @{nodes in the cluster}
+    'nodes' : type_hostname[]
+    @{internal network node names (same order as nodes)}
+    'internal' ? type_hostname[]
+};
+
+@{common device type for e.g. resource and stonith}
+type pcs_common_device = {
+    @{device options}
+    'option' ? string{}
+    @{meta options}
+    'meta' ? string{}
+    @{add to group}
+    'group' ? string
+    @{before other device}
+    'before' ? string
+    @{after other device}
+    'after' ? string
+    @{disabled}
+    'disabled' ? boolean
+    @{wait (when specified) for start in seconds}
+    'wait' ? long(0..) = 60
+};
+
+@{stonith device type}
+type pcs_stonith = {
+    include pcs_common_device
+    @{stonith device type (e.g. fence_ipmilan)}
+    'type' : string
+    @{operations (key is operation action; value are the options)}
+    'operation' ? string{}{}
+};
+
+@{resource operation type}
+type pcs_resource_operation = {
+    @{unique name (generated if not defined)}
+    'id' ? string
+    @{action name}
+    'name' ? string with match(SELF, '^(start|stop|monitor|promote|demote|notify)$')
+    @{How frequently (in seconds) to perform the operation}
+    'interval' ? long(0..)
+    @{How long to wait before declaring the action has failed }
+    'timeout' ? long(0..)
+    @{The action to take if this action ever fails}
+    'on-fail' ? string with match(SELF, '^(ignore|block|stop|restart|fence|standby)$')
+    @{If false, ignore this operation definition.}
+    'enabled' ? boolean
+    @{If true, the intention to perform the operation is recorded}
+    'record-pending' ? boolean
+    @{Run the operation only on node(s) that the cluster thinks should be in the specified role.}
+    'role' ? string with match(SELF, '^(Stopped|Started|Slave|Master)$')
+};
+
+@{resource type. special cases:
+    type = master: without standard or provider, create a named master
+        from existing resource (or group)
+        (specified by single element master dict() resource=resource_id)
+}
+type pcs_resource = {
+    include pcs_common_device
+    @{resource device standard ([standard[:provider]:]type) (e.g. ocf)}
+    'standard' ? string
+    @{resource device provider ([standard[:provider]:]type) (e.g. heartbeat)}
+    'provider' ? string
+    @{resource device type ([standard[:provider]:]type) (e.g. IPaddr2)
+      or special cases like master}
+    'type' : string
```

Comment:
No validation on any of these strings?
I assume that at a minimum they should not contain white-space.